api/pkg/service/resource/resource.go

```diff
@@ -350,7 +351,31 @@ func tinyVersionInfo(r model.ResourceVersion) *resource.ResourceVersionData {
 	return res
 }
 
-func minVersionInfo(r model.ResourceVersion) *resource.ResourceVersionData {
+func resourceHubUrlData(resId uint, req res.Request) (res.HubUrl, error) {
+	var resource model.Resource
+
+	q := req.Db.Model(&model.Resource{}).Where("id = ?", resId)
```

Comment:
don't we have resource details in the object coming from resource package where we run the query?

Reply:
Nope we don't have the resource details in the object, it is been called from `minVersionInfo` where the resourceVersion has a tiny view where it includes the id, version, rawUrl, webUrl, platforms and now hubUrl